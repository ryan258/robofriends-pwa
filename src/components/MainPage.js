import React, { Component } from 'react'

import CardList from '../components/CardList'
import SearchBox from '../components/SearchBox'
import Scroll from '../components/Scroll'
import ErrorBoundry from '../components/ErrorBoundry'

import './MainPage.css'

class MainPage extends Component {
  componentDidMount() {
    this.props.onRequestRobots()
  }

  filterRobots = () => {
    return this.props.robots.filter((robot) => {
      return robot.name.toLowerCase().includes(this.props.searchField.toLowerCase())
    })
  }

  render() {
    const { robots, onSearchChange, isPending } = this.props

    return (
      <div className="tc">
        <h1 className="f2">RoboFriends</h1>
        <SearchBox searchChange={onSearchChange} />
        <Scroll>
          {isPending ? (
            <h1>Loading</h1>
          ) : (
            <ErrorBoundry>
              <CardList robots={this.filterRobots(this.props.robots)} />
            </ErrorBoundry>
          )}
        </Scroll>
      </div>
    )
  }
}

export default MainPage
