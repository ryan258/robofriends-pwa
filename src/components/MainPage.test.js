import React from 'react'

import MainPage from './MainPage'
import { shallow } from 'enzyme'

let wrapper

// beforeEach() - runs before each test, cuts down on repetition
beforeEach(() => {
  const mockProps = {
    onRequestRobots: jest.fn(), // fake the fn call
    robots: [],
    searchField: '',
    isPending: false
  }
  wrapper = shallow(<MainPage {...mockProps} />)
})

it('renders MainPage w/o crashing', () => {
  expect(wrapper).toMatchSnapshot()
})

it('filters robots correctly', () => {
  const mockProps2 = {
    onRequestRobots: jest.fn(), // fake the fn call
    robots: [
      {
        id: 3,
        name: 'Bender',
        email: 'bender@wwf.com'
      }
    ],
    searchField: 'bender',
    isPending: false
  }
  const wrapper2 = shallow(<MainPage {...mockProps2} />)
  // .instance() will give us access to all the methods on the component being tested
  expect(wrapper2.instance().filterRobots()).toEqual([
    {
      id: 3,
      name: 'Bender',
      email: 'bender@wwf.com'
    }
  ])
})

it('filters robots correctly 2', () => {
  const mockProps3 = {
    onRequestRobots: jest.fn(),
    robots: [
      {
        id: 3,
        name: 'Bender',
        email: 'bender@wwf.com'
      }
    ],
    searchField: 'a',
    isPending: false
  }

  const filteredRobots = []

  const wrapper3 = shallow(<MainPage {...mockProps3} />)

  expect(wrapper3.instance().filterRobots()).toEqual(filteredRobots)
})
