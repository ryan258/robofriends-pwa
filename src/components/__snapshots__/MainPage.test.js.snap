// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MainPage w/o crashing 1`] = `
<div
  className="tc"
>
  <h1
    className="f2"
  >
    RoboFriends
  </h1>
  <SearchBox />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
